lib/iris/fileformats/nimrod_load_rules.py

```diff
@@ -334,29 +884,34 @@ def run(field):
         * A new :class:`~iris.cube.Cube`, created from the NimrodField.
 
     """
-    cube = iris.cube.Cube(field.data)
+    cube = iris.cube.Cube(field.data.astype(np.float32))
 
     name(cube, field)
+    mask_cube(cube, field)
     units(cube, field)
 
     # time
     time(cube, field)
     reference_time(cube, field)
+    forecast_period(cube)
 
     experiment(cube, field)
 
     # horizontal grid
-    proj_biaxial_ellipsoid(cube, field)
-    tm_meridian_scaling(cube, field)
     horizontal_grid(cube, field)
 
     # vertical
     vertical_coord(cube, field)
 
     # add other stuff, if present
-    ensemble_member(cube, field)
+    soil_type_coord(cube, field)
+    if not probability_coord(cube, field):
+        ensemble_member(cube, field)
+    time_averaging(cube, field)
     attributes(cube, field)
 
     origin_corner(cube, field)
 
+    cube.data = cube.data.astype(np.float32)
```

Comment:
This cast is effectively done twice, once at the start of this routine + once here at the end.
It may well be needed though.
Are you clear on this @MoseleyS  ?

Reply:
I'd like to leave this. The code does some unit-conversion to the data which may result in the data being recast to float64 behind the scenes, so the final `astype` is to handle this.

Reply:
:+1: